src/jvm/test/org/jetbrains/kotlinx/lincheck_test/trace_debugger/io/FileTests.kt

```diff
@@ -0,0 +1,450 @@
+
+            return readContent
+        } finally {
+            // Clean up
+            Files.delete(fileName)
+        }
+    }
+}
+
+@Ignore("java.nio.channels are not supported yet")
+class FilesAppendTest : FileOperationTest() {
+    private val fileName = Path("test-append.txt")
+
+    @Operation
+    fun operation(): String {
+        // Create a file
+        val file = Files.createFile(fileName)
+        require(Files.exists(file)) { "File was not created: $file" }
+
+        try {
+            // Write initial content
+            val initialContent = "Hello, "
+            Files.write(file, initialContent.toByteArray(), StandardOpenOption.WRITE)
+
+            // Append content
+            val appendContent = "NIO World!"
+            Files.write(file, appendContent.toByteArray(), StandardOpenOption.APPEND)
+
+            // Read full content
+            val readContent = String(Files.readAllBytes(file))
+            val expectedContent = initialContent + appendContent
+
+            // Verify content
+            require(readContent == expectedContent) { 
+                "Content mismatch: expected '$expectedContent', got '$readContent'" 
+            }
+
+            return readContent
+        } finally {
+            // Clean up
+            Files.delete(fileName)
+        }
+    }
+}
+
+// File Editing Tests - IO API
+
+class FileWriteReadTest : FileOperationTest() {
+    private val file = File("test-write-read-io.txt")
+
+    @Operation
+    fun operation(): String {
+        // Create a file
+        require(file.createNewFile()) { "Failed to create file: $file" }
+        require(file.exists()) { "File was not created: $file" }
+
+        try {
+            // Write to file
+            val content = "Hello, IO World!"
+            FileWriter(file).use { writer ->
+                writer.write(content)
+            }
+
+            // Read from file
+            val readContent = file.inputStream().reader().use { it.readText() }
+
+            // Verify content
+            require(readContent == content) { "Content mismatch: expected '$content', got '$readContent'" }
+
+            return readContent
+        } finally {
+            // Clean up
+            require(file.delete()) { "Failed to delete file: $file" }
+            require(!file.exists()) { "File was not deleted: $file" }
+        }
+    }
+}
+
+class FileAppendTest : FileOperationTest() {
+    private val file = File("test-append-io.txt")
+
+    @Operation
+    fun operation(): String {
+        // Create a file
+        require(file.createNewFile()) { "Failed to create file: $file" }
+        require(file.exists()) { "File was not created: $file" }
+
+        try {
+            // Write initial content
+            val initialContent = "Hello, "
+            FileWriter(file).use { writer ->
+                writer.write(initialContent)
+            }
+
+            // Append content
```

Comment:
Could you please also add another similar test, where file is closed and re-opened between an intial write and append.

Reply:
But you are already pointing at the such test: `FileAppendTest`. `use` opens and closes under-the-hood.